### PR TITLE
NR-89151: Apache ZooKeeper Quickstart

### PR DESCRIPTION
**Pull request by RamanaReddy8801**
# Summary

**Ticket details**: https://issues.newrelic.com/browse/NR-89151

**Description**: This PR has the new quickstart integration named "Apache ZooKeeper"

**Apache ZooKeeper Dashboard**: https://onenr.io/0MRNpVLaKQn

A concise description of the changes being introduced. Please review the pre-merge checklist section to validate this pull request is ready for review and merge. **If it is not ready, please mark the pull request as a draft.**

_The owners of this repo are not experts in the subject matter of the quickstarts. We review for the quickstart to be functional and for security risks. If you are seeking feedback on the content of the quickstart, please seek out a subject matter expert. If you are not an internal NR contributor, we can do our best to connect you with a content reviewer._

## Pre merge checklist

- [ ] Did you check you NRQL syntax? - Does it work?
- [ ] Did you include a Data source and Documentation reference?
- [ ] Are all documentation links publicly accessible?
- [ ] Did you check your descriptive content for [voice and tone](https://docs.newrelic.com/docs/style-guide/writing-strategies/voice-strategies-docs-sound-new-relic/)?
- [ ] Did you check your descriptive content for spelling and grammar errors?
- [ ] Did you review your content with a subject matter expert? (e.g. a Browser agent quickstart is reviewed with a member of the Browser Agent team)

### Dashboards

- [ ] Does the PR contain a screenshot for each of your dashboards?
- [ ] Do your screenshots show data?
- [ ] Has the [sanitization script](https://github.com/newrelic/newrelic-quickstarts/blob/main/CONTRIBUTING.md#dashboards) been run on each dashboard?

### Alerts

- [ ] Did you check that your alerts actually work?

### CODE REVIEW

**Review by mdumpati — COMMENTED** [review]
@RamanaReddy8801 Check my comments and use only NRI-Flex. No need NRI-ZooKeeper.

File: alert-policies/apache-zooKeeper/FsyncTime.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: Fsync Time
+
+description: |+
+  This alert is triggered when the fsync time exceeds 100 for 5 minutes.
```
Comment (pkosanam) [inline alert-policies/apache-zooKeeper/FsyncTime.yml:4]:
```suggestion
  This alert is triggered when the fsync time exceeds 90 milliseconds for 5 minutes.
```

File: alert-policies/apache-zooKeeper/FsyncTime.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    threshold: 100
+    # Time in seconds; 120 - 3600
+    thresholdDuration: 300
+    # How many data points must be in violation for the duration
+    thresholdOccurrences: ALL
+  - priority: WARNING
+    # Operator used to compare against the threshold.
+    operator: ABOVE
+    # Value that triggers a violation
+    threshold: 60
```
Comment (pkosanam) [inline alert-policies/apache-zooKeeper/FsyncTime.yml:27]:
```suggestion
    threshold: 90
```

File: alert-policies/apache-zooKeeper/RejectedConnection.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+name: Rejected Connection
```
Comment (pkosanam) [inline alert-policies/apache-zooKeeper/RejectedConnection.yml:1]:
```suggestion
name: Rejected Connections
```

File: alert-policies/apache-zooKeeper/RejectedConnection.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+description: |+
+  This alert is triggered when the rejected connection exceeds 5 for 5 minutes.
```
Comment (pkosanam) [inline alert-policies/apache-zooKeeper/RejectedConnection.yml:4]:
```suggestion
  This alert is triggered when the rejected connections exceeds 3 for 5 minutes.
```

File: alert-policies/apache-zooKeeper/WatchNodeChanged.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
+name: Watch Node Changed
```
Comment (pkosanam) [inline alert-policies/apache-zooKeeper/WatchNodeChanged.yml:1]:
```suggestion
name: Watch counts on ZooKeeper nodes
```

File: alert-policies/apache-zooKeeper/WatchNodeChanged.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+description: |+
+  This alert is triggered when the watch node changes exceeds 5 for 5 minutes.
```
Comment (pkosanam) [inline alert-policies/apache-zooKeeper/WatchNodeChanged.yml:4]:
```suggestion
  This alert is triggered when the watch counts on znodes exceeds 1 for 5 minutes.
```

File: quickstarts/apache-zooKeeper/config.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+slug: apache-zookeeper
+description: |
+  ## Comprehensive monitoring quickstart for Apache ZooKeeper
+  New Relic detects your Apache ZooKeeper metrics like znode count, watch count, client connections, etc.  Monitor your Apache ZooKeeper ensemble, ensuring optimal performance and troubleshooting issues in a timely manner.
+
+  ## Why monitor Apache ZooKeeper?
+   By leveraging Apache ZooKeeper monitoring by New Relic, you gain invaluable visibility into the overall health and performance of your distributed system. This empowers you to swiftly identify and address any potential issues, ensuring optimal operational efficiency.
+
+  ## What’s included in this quickstart?
+  With Apache ZooKeeper monitoring quickstart by New Relic, you can instantly achieve comprehensive observability across your entire stack, right from the start:
```
Comment (pkosanam) [inline quickstarts/apache-zooKeeper/config.yml:11]:
```suggestion
    New Relic Apache ZooKeeper monitoring quickstart provides quality out-of-the-box reporting:
```

File: quickstarts/apache-zooKeeper/config.yml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  - Dashboards (ZNode count, watch count, max latency, client connections, etc)
```
Comment (pkosanam) [inline quickstarts/apache-zooKeeper/config.yml:12]:
```suggestion
  - Dashboards (znode count, watch count, max latency, client connections, etc)
```

File: quickstarts/apache-zooKeeper/config.yml [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  - Alerts for zookeeper (Rejected Connection, Changed watch node)
```
Comment (pkosanam) [inline quickstarts/apache-zooKeeper/config.yml:13]:
```suggestion
  - Alerts for ZooKeeper (rejected connections, watch counts on ZooKeeper nodes)
```

**Review by pkosanam — CHANGES_REQUESTED** [review]

**Review by mdumpati — APPROVED** [review]
LGTM

**Pull request merged**